syntax = "proto3";

package ecommerce.user;

import "common.proto";

service UserService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
  rpc GetUsers(GetUsersRequest) returns (GetUsersResponse);
}

message GetUserRequest {
  string userId = 1;
}

message GetUserResponse {
  ecommerce.common.Status status = 1;
  UserInfo user = 2;
}

message GetUsersRequest {
  repeated string userIds = 1;
}

message GetUsersResponse {
  ecommerce.common.Status status = 1;
  repeated UserInfo users = 2;
}

message UserInfo {
  string id = 1;
  string username = 2;
  string email = 3;
  string fullName = 4;
  string role = 5;
  string createdAt = 6;
}